App/Views/SearchView.swift

```diff
@@ -0,0 +1,1238 @@
+                              let pageNumber = Int(pageValue) else { continue }
+                        
+                        if pageNumber > maxPage {
+                            maxPage = pageNumber
+                        }
+                    }
+                    self.totalPages = maxPage
+                } else if let requestedPage = requestedPage {
+                    self.currentPage = requestedPage
+                }
+            }
+        } catch {
+            print("Error scraping search results: \(error)")
+            searchState.message = "Failed to load search results"
+        }
+    }
+    
+    func toggleForumSelection(for option: ForumSelectOption) {
+        guard let index = forumSelectOptions.firstIndex(where: { $0.id == option.id }) else { return }
+
+        let isSelected = !forumSelectOptions[index].isSelected
+        forumSelectOptions[index].isSelected = isSelected
+
+        // Find all children and update their selection state
+        for i in forumSelectOptions.indices {
+            if forumSelectOptions[i].parentIDs.contains(option.id) {
+                forumSelectOptions[i].isSelected = isSelected
+            }
+        }
+    }
+    
+    func performSearch() async {
+        // Create URL components from the base URL
+        guard let baseURL = ForumsClient.shared.baseURL else {
+            searchState.message = "Invalid base URL"
+            return
+        }
+        
+        guard var components = URLComponents(url: baseURL, resolvingAgainstBaseURL: true) else {
+            searchState.message = "Invalid URL"
+            return
+        }
+        
+        // Add path and query items
+        components.path = "/query.php"
+        components.queryItems = [
+            URLQueryItem(name: "q", value: searchState.query),
+            URLQueryItem(name: "action", value: "query")
+        ] + forumSelectOptions
+            .filter(\.isSelected)
+            .map { URLQueryItem(name: "forums[]", value: $0.value) }
+        
+        // Create the final URL and query string
+        guard let url = components.url,
+              let queryString = components.percentEncodedQuery?.data(using: String.Encoding.utf8) else {
+            searchState.message = "Failed to create search URL"
+            return
+        }
+        
+        do {
+            var request = URLRequest(url: url)
+            request.httpMethod = "POST"
+            request.setValue("application/x-www-form-urlencoded", forHTTPHeaderField: "Content-Type")
+            request.httpBody = queryString
+            
+            let (data, response) = try await URLSession.shared.data(for: request)
+            
+            guard let httpResponse = response as? HTTPURLResponse,
+                  httpResponse.statusCode.isSuccessful,
+                  let htmlString = String(data: data, encoding: .utf8) else {
+                searchState.message = "Search failed"
+                return
+            }
```

Comment:
all the url/request creation and response parsing should go in `ForumsClient`. it's ok if it returns an `HTMLDocument`, don't worry about fancier parsing into structs etc. unless you really want to :)

Reply:
Done!